tokenizers/src/tokenizer/added_vocabulary.rs

```diff
@@ -315,84 +277,63 @@ impl AddedVocabulary {
     fn find_matches<'a>(
         &self,
         sentence: &str,
-        split_re: &'a MatchingSet,
+        split_trie: &'a Trie<u8>,
     ) -> Vec<(Option<u32>, Offsets)> {
         if sentence.is_empty() {
             return vec![(None, (0, 0))];
         }
 
-        let mut matches = split_re
-            .0
-            .matches(sentence)
-            .into_iter()
-            .flat_map(|idx| {
-                regex::Regex::new(&split_re.0.patterns()[idx])
-                    .unwrap()
-                    .find_iter(sentence)
-                    .map(|m| (idx, (m.start(), m.end())))
-                    .collect::<Vec<_>>()
-            })
-            .collect::<Vec<_>>();
-
-        // We sort all the matches by their start and then by their pattern id
-        matches.sort_by(
-            |(idxa, (sa, _)), (idxb, (sb, _))| {
-                if sa != sb {
-                    sa.cmp(sb)
-                } else {
-                    idxa.cmp(idxb)
+        let mut start_offset = 0;
+        let mut splits = split_trie
+            .matches(sentence.as_bytes())
+            .flat_map(|(start, stop)| {
+                let mut start = start;
+                let mut stop = stop;
+
+                let match_ = &sentence[start..stop];
+                let id = self.added_tokens_map.get(match_).unwrap_or_else(|| {
+                    self.shadow_map
+                        .get(match_)
+                        .unwrap_or_else(|| panic!("Expected to find an id for {:?}", match_))
```

Comment:
Maybe make this function return a `Result<T>`?

Reply:
I think this would go away using `aho-corasick` (it returns the ID with the match).

Reply:
Also there are other `unwrap` which would panic too.